api/v2/checluster_types.go

```diff
@@ -541,6 +546,54 @@ type PodSecurityContext struct {
 	FsGroup *int64 `json:"fsGroup,omitempty"`
 }
 
+type CheClusterGitServices struct {
+	// +optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="GitHub"
+	GitHub []GitHubService `json:"github,omitempty"`
+	// +optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="GitLab"
+	GitLab []GitLabService `json:"gitlab,omitempty"`
+	// +optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,displayName="BitBucket"
+	BitBucket []BitBucketService `json:"bitbucket,omitempty"`
+}
+
+// GitHubService enables users to work with a remote Git repository that is hosted on GitHub.
+type GitHubService struct {
+	// Kubernetes secret, that contains Base64-encoded GitHub OAuth Client id and GitHub OAuth Client secret data.
+	// The GitHub OAuth data must be stored in the Kubernetes secret in `id` and `secret` keys respectively.
+	// +kubebuilder:validation:Required
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,xDescriptors="urn:alm:descriptor:io.kubernetes:Secret"
+	SecretName string `json:"secretName"`
```

Comment:
@vinokurig why we don't we need the endpoint for GitHub? How does it work? Do we assume that's a github server if it's not among gitlab/bitbucket endpoints?

Reply:
As for the GitHub case, we don't support the self hosted GitHub server endpoint. It means that the SAAS version and the cloud version of GitHub enterprise always use the `github.com` endpoint. So we don't need to configure it yet.